authentication/azure_sp_objectid.go

```diff
@@ -3,47 +3,144 @@ package authentication
 import (
 	"context"
 	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/manicminer/hamilton/auth"
+	"github.com/manicminer/hamilton/environments"
+	"github.com/manicminer/hamilton/msgraph"
+	"github.com/manicminer/hamilton/odata"
+
 	"github.com/hashicorp/go-azure-helpers/sender"
 )
 
 func buildServicePrincipalObjectIDFunc(c *Config) func(ctx context.Context) (string, error) {
```

Comment:
we should probably make this:

```suggestion
func buildServicePrincipalObjectIDFunc(c *Config) func(ctx context.Context) (*string, error) {
```